src/learner.cc

```diff
@@ -666,6 +688,23 @@ class LearnerIO : public LearnerConfiguration {
       attributes_[kv.first] = get<String const>(kv.second);
     }
 
+    // feature names and types are saved in xgboost 1.4
+    if (learner.find("feature_names") != learner.cend()) {
```

Comment:
The use of `at` is to comply with const modifier.  `[]` operator is not `const` in `std::map`.

Reply:
Ah, you're quite right! In that case, how would you feel about either just catching `std::out_of_range` for that `at` call or storing the results of the `find` so that we're not doing this search twice?

Reply:
Thanks for the suggestion.  I updated it to use iterator returned by `find`.